cmd/talosctl/cmd/talos/kubeconfig.go

```diff
@@ -41,6 +41,60 @@ If merge flag is defined, config will be merged with ~/.kube/config or [local-pa
 				return err
 			}
 
+			var localPath string
+
+			if len(args) == 0 {
+				// no path given, use defaults
+				var err error
+
+				if merge {
+					var usr *user.User
+					usr, err = user.Current()
+
+					if err != nil {
+						return err
+					}
+
+					localPath = filepath.Join(usr.HomeDir, ".kube/config")
+				} else {
+					localPath, err = os.Getwd()
+					if err != nil {
+						return fmt.Errorf("error getting current working directory: %s", err)
+					}
+				}
+			} else {
+				localPath = args[0]
+			}
+
+			localPath = filepath.Clean(localPath)
+
+			st, err := os.Stat(localPath)
+			if err != nil {
+				if !os.IsNotExist(err) {
+					return fmt.Errorf("error checking path %q: %w", localPath, err)
+				}
+
+				err = os.MkdirAll(filepath.Dir(localPath), 0o755)
+				if err != nil {
+					return err
+				}
+			} else if st.IsDir() {
+				// only dir name was given, append `kubeconfig` by default
+				localPath = filepath.Join(localPath, "kubeconfig")
+			}
```

Comment:
I am torn on this ... IIUC, if the supplied local path is a dir, we use filename `kubeconfig` by default. It gives `local-path` dual meaning, which _could_ be confusing. What about making `local-path` a file name at all times, and if a directory with the same name exists, then error?

Reply:
I didn't want to break backwards compatibility with that, and to keep `talosctl kubeconfig` (without args) work as before. But certainly that's not I'm going to die for ;)

Reply:
Ah gotcha. It is not a blocker for me. Eventually I would like to phase that out if we decide not to just break it now.